### ISSUE TITLE
Submission Form Broken and Rough Points

### ISSUE THREAD

**Issue by smcgregor** [responsible-ai-collaborative/aiid#660]
1. it is possible to submit and update the incident form without setting an incident date or setting an existing incident. This should fail validation on the client (Formik?) and on the server (schema?).
2. Fields that are not assigned are displayed to the editor (incident_date and source_domain).
![Screen Shot 2022-05-31 at 11 28 46 PM](https://user-images.githubusercontent.com/64780/171341581-a7550cc8-de96-4b4a-aa3c-9139f0c7cd78.png)
3. The design of showing the incident title and date inline with the form is awkward. Maybe it would be better to move up the card that appears below giving the listing of previously submitted incident reports.
![Screen Shot 2022-05-31 at 11 31 15 PM](https://user-images.githubusercontent.com/64780/171341986-7789afaa-b4b8-408c-bae7-e3ad5b77d681.png)
4. Current tags are not populating in the submission edit form. I am not sure when the tags are or are not being updated from the form.
5. The submission edit form (modal) should close after the server successfully updates the server state of the submission.
6. Incident date is not set here when an incident is assigned to the submission
![Screen Shot 2022-05-31 at 11 48 16 PM](https://user-images.githubusercontent.com/64780/171344469-930b65e8-9a83-4731-94ca-70fc3abf0e23.png)

None of these are blocking, but some of them are potential pitfalls as @LKchemposer begins editing with a user account.

**Comment by cesarvarela** [comment]
I can't find any `reports` with tags 🤔 (in staging at least)

**Comment by cesarvarela** [comment]
`source_domain` is not present in the form. Should it be?

**Comment by smcgregor** [comment]
`source_domain` is derived from the URL and should not be presented to users.

`tags` is not currently used, but it will be in the weeks ahead.

**Comment by cesarvarela** [comment]
It looks like` source_domain` has been broken for a while 💩 I'll create a migration to have it up to date and update the forms to derive it from the URL.

I guess in the future, stuff like this would be avoided by #611

**Comment by smcgregor** [comment]
Whoops. Sounds good.

**Comment by cesarvarela** [comment]
> The design of showing the incident title and date inline with the form is awkward. Maybe it would be better to move up the card that appears below giving the listing of previously submitted incident reports.

I agree that the UI is awkward but I do not follow what the suggested change is.

**Comment by smcgregor** [comment]
Right now there are two different areas that are presenting potentially related incidents or reports. My suggestion is that the information that is getting presented below the incident report field should be in the card that appears showing reports related to the currently selected incident. I also think that card should be given the highest precedence and displayed at the top of the list of "related" cards.
